src/grammar.pegjs

```diff
@@ -191,23 +191,23 @@ constructorCall = 'new' __ target:qualifiedName _ args:arguments { return New(ta
 
 literal = booleanLiteral
         / numberLiteral
-        / nullLiteral
+        / Literal
         / stringLiteral
         / objectLiteral
         / closure
         / collectionLiteral
 
-booleanLiteral = value:('false' / 'true') { return BooleanLiteral(value === 'true') }
+booleanLiteral = value:('false' / 'true') { return Literal(value === 'true') }
 
-nullLiteral = 'null' { return NullLiteral }
+Literal = 'null' { return Literal(null) }
 
-stringLiteral = '"' value:( escapedChar  / [^'\\"] )* '"' { return StringLiteral(value.join('')) }
-              / "'" value:( escapedChar  / [^'\\'] )* "'" { return StringLiteral(value.join('')) }
+stringLiteral = '"' value:( escapedChar  / [^'\\"] )* '"' { return Literal(value.join('')) }
+              / "'" value:( escapedChar  / [^'\\'] )* "'" { return Literal(value.join('')) }
 escapedChar = '\\b'/'\\t'/'\\n'/'\\f'/'\\r'/'\\u'/'\\"'/"\\'"/'\\\\'
 
-numberLiteral = ('0x'/'0X') value:[0-9a-fA-F]+ { return NumberLiteral(parseInt(value.join(''), 16)) }
-              / whole:[0-9]+'.'decimals:[0-9]+ { return NumberLiteral(parseFloat(whole.join('')+'.'+decimals.join(''))) }
-              / value:[0-9]+                   { return NumberLiteral(parseInt(value.join(''), 10)) }
+numberLiteral = ('0x'/'0X') value:[0-9a-fA-F]+ { return Literal(parseInt(value.join(''), 16)) }
+              / whole:[0-9]+'.'decimals:[0-9]+ { return Literal(parseFloat(whole.join('')+'.'+decimals.join(''))) }
+              / value:[0-9]+                   { return Literal(parseInt(value.join(''), 10)) }
```

Comment:
Technically literals ARE typed. true "true" and 0 all have different responses of typeof. Perhaps we are overdoing it by removing the different nodes, but I took the advice from Babel, where they make a big point in not adding redundant field to nodes if the info can be deduced somehow. I think the fewer nodes there are, the easier it is to change them : /